src/analysis/inner_product_space/matrix_spectrum.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2022 Alexander Bentkamp. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alexander Bentkamp
+-/
+import analysis.inner_product_space.spectrum
+/-! # Spectral theory of hermitian matrices
+
+This file defines hermitian matrices (`matrix.is_hermitian`) and proves the spectral theorem for
+matrices. The proof of the spectral theorem is based on the spectral theorem for linear maps
+(`diagonalization_basis_apply_self_apply`).
+
+## Tags
+
+self-adjoint matrix, hermitian matrix, spectral theorem, diagonalization theorem
+
+-/
+
+namespace matrix
+
+variables {𝕜 : Type*} [is_R_or_C 𝕜] [decidable_eq 𝕜] {n : Type*} [fintype n] [decidable_eq n]
+variables {A : matrix n n 𝕜}
+
+open_locale matrix
+
+local notation `⟪`x`, `y`⟫` := @inner 𝕜 (pi_Lp 2 (λ (_ : n), 𝕜)) _ x y
+
+/-- A matrix is hermitian if it is equal to its conjugate transpose. On the reals, this definition
+captures symmetric matrices. -/
+def is_hermitian (A : matrix n n 𝕜) : Prop := Aᴴ = A
```

Comment:
This probably belongs next to `linear_algebra/matrix/symmetric`; or even in the same file.

Reply:
The spectral theorem should probably also be in `linear_algebra/matrix`, right? I've moved the whole file there and split it into `spectrum.lean` and `hermitian.lean`.